GeeksCoreLibrary/Components/Account/Services/AccountsService.cs

```diff
@@ -447,5 +454,199 @@ public async Task<List<RoleModel>> GetUserRolesAsync(ulong userId, bool includeP
             // Filter the roles based on the user's role IDs and return a List of the remaining rows.
             return roles.Where(role => userRoleIds.Contains(role.Id)).ToList();
         }
+
+        /// <summary>
+        /// NOTE: This method was moved from Accounts component and it should NOT be exposed to users.
+        /// Automatically logs in the user via ID. This function should never be made available publicly, only for internal usage to login after creating an account for example.
+        /// </summary>
+        /// <param name="userId">The ID of the user to login.</param>
+        /// <param name="mainUserId">The ID of the main user, if the user is logging in with a sub account.</param>
+        /// <param name="role">Used to set a custom role for the user separate of the Wiser role system</param>
+        /// <param name="extraDataForReplacements"></param>
+        /// <param name="settings"></param>
+        public async Task AutoLoginUserAsync(ulong userId, ulong mainUserId, string role, Dictionary<string,string> extraDataForReplacements, AccountCmsSettingsModel settings)
+        {
+            // Make sure we have a valid user ID.
+            if (userId <= 0)
+            {
+                logger.LogTrace("AutoLoginUser called with invalid user ID.");
+                return;
+            }
+
+            // Everything succeeded, so generate a cookie for the user and reset any failed login attempts.
+            var amountOfDaysToRememberCookie = GetAmountOfDaysToRememberCookie(settings);
+            var cookieValue = await GenerateNewCookieTokenAsync(userId, mainUserId, !amountOfDaysToRememberCookie.HasValue || amountOfDaysToRememberCookie.Value <= 0 ? 0 : amountOfDaysToRememberCookie.Value, settings.EntityType, settings.SubAccountEntityType, role);
+            await SaveGoogleClientIdAsync(userId, settings);
+
+            var offset = (amountOfDaysToRememberCookie ?? 0) <= 0 ? (DateTimeOffset?)null : DateTimeOffset.Now.AddDays(amountOfDaysToRememberCookie.Value);
+            var currentContext = httpContextAccessor.HttpContext;
+            HttpContextHelpers.WriteCookie(currentContext, settings.CookieName, cookieValue, offset, isEssential: true);
+
+            await SaveLoginAttemptAsync(true, userId, extraDataForReplacements, settings);
+        }
+
+        /// <summary>
+        /// Gets the Google Client ID from the Google Analytics cookie and saved it.
+        /// </summary>
+        /// <param name="userIdForGoogleCid">The ID of the user to save the CID for.</param>
+        /// <param name="settings"></param>
```

Comment:
The summary can be replaced with `/// <inheritdoc />`, then the summary of the interface is shown so we only need to manage this on one place.